### PR TITLE
Use osx-arm64 packages on Apple silicon

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

Previously the osx-64 packages were used because osx-arm64 was not supported. Rosetta 2 is no longer required for the conda runner on Apple silicon.

## Checklist

- [x] Merge https://github.com/nextstrain/conda-base/pull/80
- [x] Test locally: run a draft ebola workflow using this runtime
- [x] Test locally: auto-update works as expected

    ```
    NEXTSTRAIN_CONDA_BASE_PACKAGE="nextstrain-base ==20250226T220403Z" ~/.nextstrain/cli-standalone/nextstrain setup conda --force
    ./devel/venv-run nextstrain update
    ```

- [ ] Post-merge: merge https://github.com/nextstrain/docs.nextstrain.org/pull/246

### CODE REVIEW

File: nextstrain/cli/runner/conda.py [hunk 1]
```diff
@@ -471,8 +468,6 @@ def supported_os() -> bool:
         ('operating system is supported',
             supported_os()),
 
-        *test_rosetta_enabled(),
```
Comment (tsibley) [inline nextstrain/cli/runner/conda.py:474]: Can the definition of this function be removed then?

File: nextstrain/cli/runner/conda.py [hunk 2]
```diff
@@ -610,9 +605,10 @@ def package_distribution(channel: str, package: str, version: str = None, label:
 
     if (system, machine) == ("Linux", "x86_64"):
         subdir = "linux-64"
-    elif (system, machine) in {("Darwin", "x86_64"), ("Darwin", "arm64")}:
-        # Use the x86 arch even on arm (https://docs.nextstrain.org/en/latest/reference/faq.html#why-intel-miniconda-installer-on-apple-silicon)
```
Comment (tsibley) [inline nextstrain/cli/runner/conda.py:614]: Can this doc page go away too? (Later, non-blocking)

File: nextstrain/cli/runner/conda.py (hunk 1 above)
Comment (victorlin) [reply nextstrain/cli/runner/conda.py:474]: It's still used (optionally) for Docker:

https://github.com/nextstrain/cli/blob/4a00d7100eff811eab6df34db73c7f6d4196e22b/nextstrain/cli/runner/docker.py#L395

**Review by tsibley — APPROVED** [review]
LGTM. I haven't tested the changes, but trust you have.

**Comment by victorlin** [comment]
Oh good thinking, I'll check after lunch.

**Comment by victorlin** [comment]
First, I installed an earlier version so that `nextstrain update` will pull `20250205T180501Z` [which has an osx-arm64 variant](https://anaconda.org/Nextstrain/nextstrain-base/files?version=20250205T180501Z).

```
NEXTSTRAIN_CONDA_BASE_PACKAGE="nextstrain-base ==20250203T212457Z" nextstrain setup --force conda
```

Then I ran `nextstrain update conda` with this PR's changes. It ran fine and pulled the `osx-64` variant. This makes sense, because while `package_distribution()` will detect `arch=arm64` and return a value based on that, the code path taken [only takes the version tag](https://github.com/nextstrain/cli/blob/e274a7cfebbf79000827f594031417915fefea7b/nextstrain/cli/runner/conda.py#L500), which is platform-agnostic, and passes it to `micromamba update`.

All this to say, this is not a breaking change as long as we continue to provide both `osx-arm64` and `osx-64` variants in conda-base. Having done this testing, I've added a changelog entry:

> The Conda runtime now supports native installation for Apple silicon (e.g. M1
  chip) devices, which improves performance when using the runtime. To take full
  advantage of this feature, force a fresh installation with `nextstrain setup
  --force conda`.



<details>

<summary>
Full output of `nextstrain update conda`
</summary>

```
Checking for newer versions of Nextstrain CLI…

nextstrain-cli is up to date!

Updating conda runtime…
Updating Conda package nextstrain-base from 20250203T212457Z to 20250205T180501Z…

Updating Conda packages in /Users/victor/.nextstrain/runtimes/conda/env…
  - nextstrain-base ==20250205T180501Z
bioconda/noarch (check zst)                         Checked  0.1s
bioconda/osx-64 (check zst)                         Checked  0.1s
nextstrain/noarch (check zst)                       Checked  0.2s
nextstrain/osx-64 (check zst)                       Checked  0.3s
bioconda/osx-64                                      4.1MB @  14.6MB/s  0.3s
nextstrain/osx-64                                   39.4kB @ 135.1kB/s  0.3s
nextstrain/noarch                                  116.0 B @ 258.0 B/s  0.6s
bioconda/noarch                                      4.7MB @   5.9MB/s  0.5s
conda-forge/noarch                                  19.0MB @  19.0MB/s  1.0s
conda-forge/osx-64                                  36.4MB @  29.0MB/s  1.3s
Transaction

  Prefix: /Users/victor/.nextstrain/runtimes/conda/env

  Updating specs:

   - nextstrain-base==20250205T180501Z


  Package                     Version  Build              Channel          Size
─────────────────────────────────────────────────────────────────────────────────
  Upgrade:
─────────────────────────────────────────────────────────────────────────────────

  - awscli                    2.23.11  py311h6eed73b_0    conda-forge      13MB
  + awscli                    2.23.13  py311h6eed73b_0    conda-forge      13MB
  - paramiko                    3.5.0  pyhd8ed1ab_1       conda-forge     161kB
  + paramiko                    3.5.1  pyhd8ed1ab_0       conda-forge     161kB
  - yte                         1.5.6  pyha770c72_0       conda-forge      14kB
  + yte                         1.5.7  pyha770c72_0       conda-forge      14kB
  - nextstrain-base  20250203T212457Z  h0dc7051_0_locked  nextstrain       13kB
  + nextstrain-base  20250205T180501Z  h0dc7051_0_locked  nextstrain       12kB

  Summary:

  Upgrade: 4 packages

  Total download: 13MB

─────────────────────────────────────────────────────────────────────────────────



Transaction starting
paramiko                                           161.0kB @   2.5MB/s  0.1s
yte                                                 14.4kB @ 165.2kB/s  0.1s
awscli                                              13.3MB @  71.6MB/s  0.2s
nextstrain-base                                     12.4kB @  20.9kB/s  0.6s
Changing awscli-2.23.11-py311h6eed73b_0 ==> awscli-2.23.13-py311h6eed73b_0
Changing paramiko-3.5.0-pyhd8ed1ab_1 ==> paramiko-3.5.1-pyhd8ed1ab_0
Changing yte-1.5.6-pyha770c72_0 ==> yte-1.5.7-pyha770c72_0
Changing nextstrain-base-20250203T212457Z-h0dc7051_0_locked ==> nextstrain-base-20250205T180501Z-h0dc7051_0_locked

Transaction finished

To activate this environment, use:

    micromamba activate /Users/victor/.nextstrain/runtimes/conda/env

Or to execute a single command in this environment, use:

    micromamba run -p /Users/victor/.nextstrain/runtimes/conda/env mycommand

Cleaning up…
Collect information..
Cleaning index cache..
Cleaning lock files..
  Package file                                                 Size
─────────────────────────────────────────────────────────────────────
  /Users/victor/.nextstrain/runtimes/conda/micromamba/pkgs
─────────────────────────────────────────────────────────────────────

  awscli-2.23.13-py311h6eed73b_0.conda                         13MB
  nextstrain-base-20250205T180501Z-h0dc7051_0_locked.conda     12kB
  paramiko-3.5.1-pyhd8ed1ab_0.conda                           161kB
  yte-1.5.7-pyha770c72_0.conda                                 14kB

  /Users/victor/.nextstrain/runtimes/conda/.mamba/pkgs
─────────────────────────────────────────────────────────────────────


  
─────────────────────────────────────────────────────────────────────

  Total size:                                                  13MB
Cleaning tarballs..
  Package folder                                         Size
───────────────────────────────────────────────────────────────
  /Users/victor/.nextstrain/runtimes/conda/micromamba/pkgs
───────────────────────────────────────────────────────────────

  awscli-2.23.13-py311h6eed73b_0                        127MB
  nextstrain-base-20250205T180501Z-h0dc7051_0_locked     66kB
  paramiko-3.5.1-pyhd8ed1ab_0                           844kB
  yte-1.5.7-pyha770c72_0                                 42kB

  /Users/victor/.nextstrain/runtimes/conda/.mamba/pkgs
───────────────────────────────────────────────────────────────


  
───────────────────────────────────────────────────────────────

  Total size:                                           128MB
Cleaning packages..
warning  libmamba This does not check for packages installed using
    symlinks back to the package cache.

Runtime updated!
```

</details>

**Comment by tsibley** [comment]
> All this to say, this is not a breaking change as long as we continue to provide both `osx-arm64` and `osx-64` variants in conda-base.

An edge case here is that `nextstrain check-setup conda` will no longer test Rosetta 2 even for Conda runtime setups that are still using `osx-64` on arm64 hardware. But I think that's ok.

**Comment by tsibley** [comment]
We're gonna have to remember for a long while into the future that not all Conda runtimes on macOS will be arm64, if they've been long-lived and simply updated continually. I think it'd be simpler to reason about if we explicitly switched setups over from `osx-64` → `osx-arm64`. But that's a bit more work to get right too.

**Comment by victorlin** [comment]
Good points. How about a deprecation message like this? 76543e89bf6165c1c606d6fc705048958d185255

```
Checking for newer versions of Nextstrain CLI…

nextstrain-cli is up to date!

Updating conda runtime…

Your Conda runtime was created before version X.X.X, so it is using
Rosetta 2 for emulation. In a future major version release,
Nextstrain CLI will stop supporting Conda runtimes created before
version X.X.X. To re-create your runtime, force a fresh installation
with `nextstrain setup conda --force`.

…
```

The same code that detects "variant" of a Conda runtime setup (by inspecting installed packages, maybe there is a better way) can also be used to keep checking Rosetta 2 for `osx-64` setups on `arm64` hardware.

File: CHANGES.md [hunk 3]
```diff
@@ -13,6 +13,12 @@ development source code and as such may not be routinely kept up to date.
 
 # __NEXT__
 
+## Improvements
+
+* The Conda runtime now supports native installation for Apple silicon (e.g. M1
+  chip) devices, which improves performance when using the runtime. To take full
+  advantage of this feature, force a fresh installation with `nextstrain setup
+  --force conda`.
```
[the remark this answers is not in the document]
Comment (victorlin) [reply CHANGES.md]: [Updated](https://github.com/nextstrain/cli/blob/37046482d3ff8cd074d22e2bea8dcbfa32d03b89/CHANGES.md?plain=1#L18-L21).

**Comment by tsibley** [comment]
> How about a deprecation message like this?

For a deprecation message, that's good. But I'm wondering why we would "stop supporting Conda runtimes created before version X.X.X" (would it be merely to remove the subdir detection and conditional Rosetta 2 check?) and why we'd ask the user to recreate the runtime instead of doing it automatically on update?

My thinking was we could add code to `update()` to automatically re-setup the runtime fresh if updating `nextstrain-base` to a version `>= "20250203T212457Z"` when the currently-installed version's subdir is `osx-64`. (The latter can be read out of installed package metadata, and there's a helper function for that already.)

**Comment by victorlin** [comment]
That makes sense, I'll look into updating this PR with that behavior.

**Review by genehack — APPROVED** [review]

**Comment by victorlin** [comment]
> That makes sense, I'll look into updating this PR with that behavior.

Almost 2 months later... done as dc275b985d0723af9a6e2a5e84176ab8bfe41790 (needed a refactor, c9891d208a48d526a7f9323f9440e2744f055c9a).

File: nextstrain/cli/runner/conda.py (hunk 2 above)
Comment (victorlin) [reply nextstrain/cli/runner/conda.py:609]: https://github.com/nextstrain/docs.nextstrain.org/pull/246

File: nextstrain/cli/runner/conda.py [hunk 4]
```diff
@@ -524,6 +541,13 @@ def update() -> RunnerUpdateStatus:
     print(f"Updating Conda packages in {PREFIX}…")
     print(f"  - {update_spec}")
 
+    if osx_arm64_emulated():
+        requested_version = PackageSpec.parse(update_spec).version_spec or ""
```
Comment (tsibley) [inline nextstrain/cli/runner/conda.py:545]: One issue that stands out here is that `version_spec` is not a bare version, e.g. it's often something like `==XYZ`. Comparing it with `>=` against a bare version won't work as this code expects.

**Review by tsibley — CHANGES_REQUESTED** [review]
Thanks! I spotted a couple problems (one noted below) and also wanted to adjust things for some code-organization preferences I have, so I found myself just hacking on things instead of filing a bunch of review comments. I'll push up changes and see what you think?

**Comment by victorlin** [comment]
Thanks @tsibley, please do!

File: nextstrain/cli/runner/conda.py (hunk 4 above)
Comment (victorlin) [reply nextstrain/cli/runner/conda.py:545]: Oof, I should have done more testing...

**Comment by tsibley** [comment]
Noting for posterity that I'd said `>= 20250203T212457Z` but it should actually be `> 20250203T212457Z` or `>= 20250203T233208Z`.

**Comment by victorlin** [comment]
Hmm, I just tried using the osx-arm64 package on an M1 Mac (home computer) and cannot get `augur` to run on any of the available versions (`>= 20250205T180501Z`): https://github.com/nextstrain/conda-base/issues/106

My testing notes show that I successfully ran an ebola workflow using the new package, which should have hit this error, but it's possible that was done on another machine. We may want to look into this before merge.

**Comment by tsibley** [comment]
> […] so I found myself just hacking on things instead of filing a bunch of review comments. I'll push up changes and see what you think?

See https://github.com/nextstrain/cli/pull/436, which is waaaay more than my initial "review comments" scope. :sweat_smile: 

My "just hacking on things" turned into more than I anticipated because I took a detour into resolving the [use-of-anaconda.org issue](https://github.com/nextstrain/cli/issues/422) since it was an annoying blocker to testing on `blab-mac` and on my own machine when at lab. I also took another detour elsewhere to do https://github.com/nextstrain/conda-base/pull/111 and https://github.com/nextstrain/conda-base/pull/114, which led to reworking some of my initial hacking here to account for my newly acquired, much deeper understanding of builds in Conda's model.

**Comment by victorlin** [comment]
#436 looks good. Closing in favor of that.
